pandas/tests/window/moments/test_moments_rolling_apply.py

```diff
@@ -122,13 +122,16 @@ def test_center_reindex_series(raw, series):
     s = [f"x{x:d}" for x in range(12)]
     minp = 10
 
-    series_xp = (
-        series.reindex(list(series.index) + s)
-        .rolling(window=25, min_periods=minp)
-        .apply(f, raw=raw)
-        .shift(-12)
-        .reindex(series.index)
-    )
+    warn = None if raw else FutureWarning
+    with tm.assert_produces_warning(warn, check_stacklevel=False):
+        # GH#36697 is_all_dates deprecated
+        series_xp = (
+            series.reindex(list(series.index) + s)
+            .rolling(window=25, min_periods=minp)
+            .apply(f, raw=raw)
+            .shift(-12)
+            .reindex(series.index)
```

Comment:
Why is this case raising a warning?

Reply:
ping

Reply:
`series.reindex(list(series.index) + s)` has an object-dtype Index that can be cast to DatetimeIndex

Reply:
@jbrockmendel Thanks for the follow-up.

Now, the user is not actively creating a Seriew with an object-dtype index themselves, but it is happening under the hood in the rolling apply. 
So I am not sure the user should see this warning (and it's actually not a single warning, but a flood of warnings, one for each window ..). Or what is the recommended way in this case to get rid of the warning?

Reply:
could suppress it inside roll_apply?

Reply:
Possibly. The warning only shows with `raw=False`, so I suppose this comes from creating a Series for each window. We might also be able to avoid the warning altogether (without the need to suppress it) by modifying the series construction to be explicit about the index dtype?

Reply:
IMO we should address this before the release

Reply:
I opened https://github.com/pandas-dev/pandas/issues/38182 for this